ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/answers/activity-file-answer/activityFileAnswer.component.html

```diff
@@ -1,27 +1,51 @@
+<!--  TODO: Add i18n  -->
 <ddp-question-prompt [block]="block"></ddp-question-prompt>
-<div class="file-upload-content">
+
+<div class="file-requirements" *ngIf="fileMaxSize || fileMimeTypes.length">
+    <div class="file-max-size" *ngIf="fileMaxSize">Max size: {{fileMaxSize}}</div>
+    <div class="file-types" *ngIf="fileMimeTypes.length">Allowed mime-types: {{fileMimeTypes.join(', ')}}</div>
+</div>
+
+<div class="file-upload-content" [class.readonly]="readonly">
     <input type="file" class="file-input"
+           [accept]="fileMimeTypes.join(',')"
            (change)="onFilesSelected($event.target.files)" #fileUpload>
 
     <div class="file-select">
-        <div class="drop-block" dropFileToUpload (filesDropped)="onFilesSelected($event)">
+        <div class="drop-block"
+             dropFileToUpload
+             (filesDropped)="!readonly && onFilesSelected($event)">
             <mat-icon class="upload-icon" color="primary">file_upload</mat-icon>
             <span class="drop-block-text">Drag files to upload</span>
         </div>
         <button mat-raised-button color="primary" class="upload-btn"
+                [disabled]="readonly"
                 (click)="fileUpload.click()">Choose file
             <mat-icon>attach_file</mat-icon>
         </button>
 
+        <div class="selected-file">{{selectedFile?.name}}</div>
+
+        <button *ngIf="!readonly" #uploadBtn
+                mat-raised-button color="primary"
+                class="submit-btn"
+                [disabled]="!isFileSelected"
+                (click)="submitFileUpload()">{{uploadedFile?.isUploaded? 'Reupload' : 'Submit'}}
+        </button>
     </div>
-    <div class="file-info">
-        <span class="file-name">{{file?.name || 'No file uploaded'}}</span>
+
+    <div class="uploaded-file">
+        <div class="title">Uploaded:</div>
```

Comment:
**_As for i18n:_**
I have `TODO: add i18n` at the top of the component.
The thing is that we don't know for now where the File Upload functionality will be applied (in which activity).
It does not make sense to apply FileUpload translations for all existing studies right now, in my opinion. 
I have added translations for modal dialog because it's already used and translated in some studies. 

What do you think about it?

Reply:
Not needed in any of our current studies. But coming soon!